src/main/scala/chisel3/aop/Select.scala

```diff
@@ -65,9 +65,11 @@ object Select {
   /** Selects all Instances of instances/modules directly instantiated within given module, of provided type
     *
     * @note IMPORTANT: this function requires summoning a TypeTag[T], which will fail if T is an inner class.
+    * @note IMPORTANT: this function ignores type parameters. E.g. instancesOf[List[Int]] would return List[String].
+    * 
     * @param parent hierarchy which instantiates the returned Definitions
     */
-  def instancesOf[T <: BaseModule : TypeTag](parent: Hierarchy[BaseModule]): Seq[Instance[T]] = {
+  def instancesOf[T <: BaseModule](parent: Hierarchy[BaseModule])(implicit tag: TypeTag[T]): Seq[Instance[T]] = {
```

Comment:
oops this was just for debugging will roll back this line

Reply:
I'll also note that you don't need to desugar to get the typeclass instance. It is equivalent to write `val tag = implicitly[TypeTag[T]]` as a line in the function. Just a helpful tip whenever someone is using the ` : Typeclass` syntax.